proxy/rewrites.go

```diff
@@ -3,20 +3,26 @@ package proxy
 import (
 	"bytes"
 	"compress/gzip"
+	"github.com/pkg/errors"
 	"io"
 	"net/http"
 	"net/url"
 	"path"
 	"strings"
-
-	"github.com/pkg/errors"
 )
 
 type compressableBody struct {
 	buf bytes.Buffer
 	w   io.WriteCloser
 }
 
+// we require a read and write for websocket connections
+var _ io.ReadWriteCloser = new(compressableBody)
+
+func (b *compressableBody) Close() error {
+	return nil
```

Comment:
Shouldn't this close the inner elements as to avoid memory / handler leaks? i.e.

```
b.w.Close()
b.buf.Clean() // or whatever the method is called the empties a buffer :)
```

Reply:
I'll look into this :)

Reply:
The only thing I could think of was to reset the buffer entirely, but it has no `Close()` function.

```go
func (b *compressableBody) Close() error {
	if b != nil {
		b.buf.Reset()
	}
	return nil
}
```

Reply:
I added a close for this, we initially returned a https://pkg.go.dev/io#NopCloser.